### PR TITLE
Adaptive pruning feature - cmf messages

### PR DESCRIPTION
**Pull request by titusen**
addition of the PruneSwitchModeRequest  message and alteration to existing ones.

### CODE REVIEW

File: reconfiguration/cmf/concord.cmf [hunk 1]
```diff
@@ -55,105 +55,112 @@ Msg PruneStatus 18 {
...
 
-Msg InstallStatusResponse 22 {
+Msg InstallStatusResponse 23 {
     string version
     bool in_progress
     bool install_completed
 }
 
-Msg GetVersionResponse 23 {
+Msg GetVersionResponse 24 {
```
Comment (teoparvanov) [inline reconfiguration/cmf/concord.cmf:85]: What's the impact of shifting all the message id-s? Doesn't it break backwards-compatibility?

Comment (teoparvanov) [reply reconfiguration/cmf/concord.cmf:85]: AFAIK, you should add new messages at the end, without changing existing IDs.

File: reconfiguration/cmf/concord.cmf [hunk 2]
```diff
@@ -55,105 +55,112 @@ Msg PruneStatus 18 {
     uint64 sender
     bool in_progress
     uint64 last_pruned_block
+    string configuration_json
 }
 
-Msg GetVersionCommand 19 {
+Msg PruneSwitchModeRequest 19 {
+    uint64 sender_id
+    string configuration_json
```
[the remark this answers is not in the document]
Comment (titusen) [reply reconfiguration/cmf/concord.cmf:63]: e.g. 
{ "mode" : "adaptive",
"max" : {"db": 400 , "consensus":800 }}

File: reconfiguration/cmf/concord.cmf [hunk 3]
```diff
@@ -247,6 +248,11 @@ Msg ClientReconfigurationStateReply 49 {
     list ClientStateReply states
 }
 
+Msg PruneSwitchModeRequest 50 {
+    uint64 sender_id
+    string configuration_json
```
Comment (cloudnoize) [inline reconfiguration/cmf/concord.cmf:253]: Who will compose this JSON? the operator?
Maybe it's better to stay in the CMF world and to define a new message for the configuration like
Msg PruneConfiguration{
        uint8 mode
        int rate
}

And then to compose it in that msg, this is how we work with other compositions.

Comment (teoparvanov) [reply reconfiguration/cmf/concord.cmf:253]: +1, I'd also prefer that - no need to mix CMF and JSON I believe

File: reconfiguration/cmf/concord.cmf (hunk 1 above)
Comment (titusen) [reply reconfiguration/cmf/concord.cmf]: Done

File: reconfiguration/cmf/concord.cmf (hunk 3 above)
Comment (titusen) [reply reconfiguration/cmf/concord.cmf:253]: IMO it couples message format with pruning mechanism. Let's say that there are 10 configurations of adaptive pruning. R we going to have 10 messages for each of them?

Comment (teoparvanov) [reply reconfiguration/cmf/concord.cmf]: You would have this logical coupling regardless of the format. If you use JSON, the coupling would simply be hidden as a string.

**Review by teoparvanov — APPROVED** [review]
LGTM, thanks Tytus!

**Pull request merged**